sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -242,6 +242,12 @@ private int CompareHeaderDictionaries(SortedDictionary<string, string[]> headers
                 var requestHeaderValues = header.Value;
                 var headerName = header.Key;
 
+                if (!_compareBodies && "Content-Length".Equals(headerName, StringComparison.OrdinalIgnoreCase))
```

Comment:
You can achieve the same by adding `Content-Length` to ignored headers.

Reply:
But I believe this is a bug in the existing `RecordMatcher`. If we opted to not compare bodies, it shouldn't be comparing `Content-Length`.